### PR TITLE
AudioResponsive Palette

### PR DESCRIPTION
**Pull request by netmindz**
Initial code trying to manually port the AudioResponsive Palette from MM to SR fork

### CODE REVIEW

**Comment by netmindz** [comment]
Compiles now thanks to @softhack007 

Not tried flashing on a device device as don't have one to hand that's still on the SR branch

**Comment by netmindz** [comment]
Code complete, just needs someone to check real output works, looking good on peek

**Comment by softhack007** [comment]
Thanks 👍 I'll give it a try on the weekend.

File: wled00/util.cpp [hunk 1]
```diff
@@ -66,3 +66,27 @@ uint8_t extractModeName(uint8_t mode, const char *src, char *dest, uint8_t maxLe
...
+CRGB getCRGBForBand(int x, int pal) { 
+  extern int fftResult[];                         // summary of bins array. 16 summary bins.
+  CRGB value;
+  CHSV hsv;
+  if(pal == 71) { // bit hacky to use palette id here, but don't want to litter the code with lots of different methods. TODO: add enum for palette creation type
+    if(x == 0) {
+      value = CRGB(fftResult[10]/2, fftResult[4]/2, fftResult[0]/2);
+    }
+    else if(x == 255) {
+      value = CRGB(fftResult[10]/2, fftResult[0]/2, fftResult[4]/2);
```
Comment (softhack007) [inline wled00/util.cpp:80]: fftResult[] is int in SR 0.13.x - so better to cast all CRGB parameters to uint8_t

Like `value = CRGB(uint8_t(fftResult[10]/2), uint8_t(fftResult[4]/2), uint8_t(fftResult[0]/2));`

File: wled00/util.cpp [hunk 2]
```diff
@@ -66,3 +66,27 @@ uint8_t extractModeName(uint8_t mode, const char *src, char *dest, uint8_t maxLe
...
+    } 
+    else {
+      value = CRGB(fftResult[0]/2, fftResult[4]/2, fftResult[10]/2);
+    } 
+  }
+  else if(pal == 72) {
+    int b = map(x, 0, 255, 0, 8); // convert palette position to lower half of freq band
+    hsv = CHSV(fftResult[b], 255, map(fftResult[b], 0, 255, 30, 255));  // pick hue
```
Comment (softhack007) [inline wled00/util.cpp:88]: same here - better add cast to (uint8_t) for the first parameter of CHSV

**Comment by softhack007** [comment]
Hi, looks good already 👍 
Just check that all parameters for CHSV() are cast to uint8_t, to be sure the compiler selects the right constructor.

File: wled00/util.cpp [hunk 3]
```diff
@@ -66,3 +66,27 @@ uint8_t extractModeName(uint8_t mode, const char *src, char *dest, uint8_t maxLe
...
+  CHSV hsv;
+  if(pal == 71) { // bit hacky to use palette id here, but don't want to litter the code with lots of different methods. TODO: add enum for palette creation type
+    if(x == 0) {
+      value = CRGB(fftResult[10]/2, fftResult[4]/2, fftResult[0]/2);
+    }
+    else if(x == 255) {
+      value = CRGB(fftResult[10]/2, fftResult[0]/2, fftResult[4]/2);
+    } 
+    else {
+      value = CRGB(fftResult[0]/2, fftResult[4]/2, fftResult[10]/2);
```
[the remark this answers is not in the document]
Comment (netmindz) [reply wled00/util.cpp:83]: Helps prevent being too pastel colours if memory serves correctly

[the remark this answers is not in the document]
Comment (softhack007) [reply wled00/util.cpp:83]: ah ok, makes sense 👍

**Comment by netmindz** [comment]
Think I've done all your proposed changes @softhack007

**Comment by softhack007** [comment]
thanks, then I'll merge it 👍

**Pull request merged**